modules/upgrade/pages/db-migration-12.adoc

```diff
@@ -0,0 +1,123 @@
+. Check the active smdba version:
++
+----
+rpm -q smdba
+----
++
+// CHECKIT!
++
+PostgreSQL{nbsp}10 requires ``smdba`` version 1.6.2 or later.
+
+. Perform a database backup.
+For more information on backing up, see xref:administration:backup-restore.adoc[].
+
+
+
+[[db-migration-12-upgrade]]
+== Upgrade PostgreSQL
+
+[WARNING]
+====
+Always create a database backup before performing a migration.
+For more information about a database backup, see xref:db-migration.adoc#db-migration-prepare[].
+====
+
+PostgreSQL upgrades can be performed in two ways: a regular upgrade, or a fast upgrade:
+
+A regular upgrade will create a complete copy of the database, so you will need double the existing database size of space available.
+Regular upgrades can take a considerable amount of time, depending on the size of the database and the speed of the storage system.
+
+A fast upgrade only takes a few minutes, and uses almost no additional disk space.
+However, if a fast upgrade fails, you must restore the database from the backup.
+A fast upgrade reduces the risk of running out of disk space.
+A regular upgrade will copy the database files instead of creating hard links between the files.
+
+PostgreSQL stores data at [path]``/var/lib/pgsql/data/``.
+
+.Procedure: Performing a Regular Upgrade
+. Perform a database backup.
+For more information on backing up, see xref:administration:backup-restore.adoc[].
+. Start the upgrade:
++
+----
+/usr/lib/susemanager/bin/pg-migrate-10-to-12.sh
+----
+. When the upgrade has successfully completed, you can safely delete the old database directory and reclaim lost disk space.
+The old directory is renamed to [path]``/var/lib/pgsql/data-pg10``.
++
+// CHECKIT! ====>>>====>>>=======================================> ^^^^
+
+The [path]``pg-migrate-10-to-12.sh`` script performs these operations:
+
+* Stop spacewalk services
+* Shut down the running database
+* Check if PostgreSQL{nbsp}12 is installed and install it if necessary
+* Switch from PostgreSQL{nbsp}10 to PostgreSQL{nbsp}12 as the new default
+* Initiate the database migration
+* Create a PostgreSQL configuration file tuned for use by {productname}
+* Start the database and spacewalk services
+
+[NOTE]
+====
+If the upgrade fails, the migration script will attempt to restore the database to its original state.
+====
+
+.Procedure: Performing a Fast PostgreSQL Upgrade
+. Perform a database backup.
+Without a verified database backup, you must not initiate a fast upgrade.
+For more information on backing up, see xref:administration:backup-restore.adoc[].
+. Start the upgrade:
++
+----
+/usr/lib/susemanager/bin/pg-migrate-10-to-12.sh fast
+----
+. When the upgrade has successfully completed, you can safely delete the old database directory and reclaim lost disk space.
+The old directory is renamed to [path]``/var/lib/pgsql/data-pg10``.
+// CHECKIT! ====>>>====>>>=======================================> ^^^^
```

Comment:
@mantel is this needed in case of "fast" update?

Reply:
Yes, the upgrade will just use hard links instead of copies. At least this is how it was in the past.

Reply:
thanks for feedback, Hubert!